### PR TITLE
refactor duplicates in tests

### LINKED ISSUE #417: Rewrite "RSpec"y Context/Should tests into before_all/test
It's a bit disappointing that this split in style crept in at all, but I've been seeing a lot of flakiness with flapping tests, surprising and unintuitive cases of repeated work slowing down tests, etc, stemming from all of the crufty magic this DSL performs (quick: describe the concrete behaviour of nested `context` blocks with multiple `setup`s)

(example weirdness I've seen from this nested context stuff: I've had assertions fail on the line _after_ a `binding.pry` I inserted to find out why it was failing, without the `binding.pry` activating first. This stuff is really, really fragile and hard to work with)

In contrast the `before_all`/`test 'test name' do` behaviour is straightforward and easy to reason about. I think we should standardize on it.

### PR DESCRIPTION
**Pull request by pgwillia**
Closes #417 for now.

From `gem install flay && flay test > duplicates.txt`

- [x] JOHNDOE_AUTH_HASH
- [x] removed second @item1
- [x] Random Item Fixture
  test/models/file_set_test.rb:35
  test/models/item_test.rb:331
- [x] Fancy Item Fixture
  test/controllers/file_sets_controller_test.rb:13
  test/integration/sitemap_test.rb:12
  test/integration/sitemap_test.rb:43
- [x] Nice Community and Collection fixture
  test/controllers/admin/collections_controller_test.rb:5
  test/controllers/collections_controller_test.rb:5
- [x] Books Community and Collection fixture
  test/controllers/items/files_controller_test.rb:6
  test/system/deposit_item_test.rb:5
- [x] Books Community and Collection fixture
  test/controllers/items/draft_controller_test.rb:5
  test/models/draft_item_test.rb:5
- [x] Fancy Community & Collection fixture, maybe Fancy Item fixture
  test/system/item_show_test.rb:15
  test/system/item_show_test.rb:36
- [x] Community and foo Collection fixture, maybe Random Item fixture?
  test/models/file_set_test.rb:32
  test/models/item_test.rb:328

### CODE REVIEW

File: test/system/authentication_test.rb [hunk 1]
```diff
@@ -2,19 +2,20 @@
 
 class AuthenticationTest < ApplicationSystemTestCase
 
+  JOHNDOE_AUTH_HASH = OmniAuth::AuthHash.new(
```
Comment (murny) [inline test/system/authentication_test.rb]: Nice refactor :+1:

File: test/ldp_fixtures/collection.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+nice:
```
Comment (murny) [inline test/ldp_fixtures/collection.yml]: Did you see any reason why we need so many versions of different collections/communities? Or can we be reusing the same ones over and over again (remove a few of these and reuse one or two instead)?

Comment (pgwillia) [reply test/ldp_fixtures/collection.yml]: Some have subtle differences in [visibility](https://github.com/ualbertalib/jupiter/blob/c17b8976ba6c88d11861a11fb1c5c4e767e01303/test/models/thesis_test.rb#L10) [fedora3_uuid](https://github.com/ualbertalib/jupiter/blob/master/test/controllers/redirect_controller_test.rb#L14), or [owner](https://github.com/ualbertalib/jupiter/blob/b0e18d3c6cd99cc684d1e931d623624c25854f51/test/models/collection_test.rb#L7), etc. For the ones that are there, for sure there only really needs to be one or two. I guess just laziness on my part.

**Comment by murny** [comment]
That thought also crossed my mind, but looking at travisCI runs, it doesn't seem all that much slower? Did you do any bench marking to see the difference of performance? I

t's probably fine how it is. Going off TravisCI build times doesn't seem that big of a difference. (even if its a bit slower, this helps future proof us with the migration path off fedora etc).

**Pull request merged**